mdbook/src/15-interrupts/README.md

```diff
@@ -0,0 +1,83 @@
+waits. On top of that, a CPU core continuously polling a peripheral wastes power, and in a lot of
+applications, we can't have that. Can we do better?
+
+Luckily, we can. While our little microcontroller can't compute things in parallel, it can easily
+switch between different tasks during execution, responding to events from the outside world. This
+switching is done using a feature called "interrupts"!
+
+Interrupts are aptly named: they allow peripherals to actually interrupt the core program execution
+at any point in time. On our MB2's nRF52833, peripherals are connected to the core's Nested Vectored
+Interrupt Controller (NVIC). The NVIC can stop the CPU in its tracks, instruct it to go do something
+else, and once that's done, get the CPU back to what it was doing before it was interrupted. We'll
+cover the Nested and Vectored parts of the interrupt controller later: let's first focus on how the
+core switches tasks.
+
+### Handling Interrupts
+
+Computation is always contextual: the core always needs memory to load inputs and store outputs to.
+Our microcontroller is of what's known as a load-store-architecture, and as such the core does not
+store and load it's computation parameters and results in RAM directly.  Instead, our core has
+access to a small amount scratch pad memory: the CPU registers.  Confusingly, these CPU registers
+are different from the device registers we discussed earlier in the [Registers] chapter.
+
+As far as the core is concerned, all context about the computation that it is doing is stored in the
+CPU registers. If the core is going to switch tasks, it must store the contents of the CPU registers
+somewhere, so that the new task can use them as their own scratchpad memory.  Sure enough, that is
+exactly the first thing the core does in response to an interrupt request: it stops what it's doing
+immediately and stores the contents of the CPU registers on the stack.
+
+The next step is actually jumping to the code that should be run in response to an interrupt.
+Interrupt Service Routines (ISRs), often referred to as interrupt handlers, are special functions in
+your application code that get called by the core in response to specific interrupts. An ISR
+function "returns" using a special return-from-interrupt machine instruction that causes the CPU to
+restore the CPU registers and jump back to where it was before the ISR was called.
+
+## Poke The MB2
+
+Let's define an ISR and configure an interrupt to "poke" the MB2 when Button A is pressed
+(`examples/poke.rs`). The board will respond by saying "ouch" and panicking.
+
+```rust
+{{#include examples/poke.rs}}
+```
+
+The ISR handler function is "special". The name `GPIOTE` is required here, and the function must be
+decorated with `#[interrupt]` so that it returns using a return-from-interrupt instruction rather
+than the normal way. The function may not take arguments and must return `()`.
```

Comment:
I'm kind of missing the important detail here that this #[interrupt] does three things (not just one):

1. It rewrites the function to make be an interrupt handler (including the return from interrupt)
2. It renames the function to be not callable from the Rust program
3. It adds it to the interrupt table of the program so the processor knows which function to execute